bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/image/ImageCatalogClient.scala

```diff
@@ -0,0 +1,266 @@
+package edu.gemini.catalog.image
+
+import java.io._
+import java.net.URL
+import java.nio.file.{Files, Path, StandardCopyOption}
+import java.util.logging.Logger
+
+import edu.gemini.spModel.core.{Angle, Coordinates, Declination, RightAscension}
+import nom.tam.fits.{Fits, Header}
+
+import scala.util.matching.Regex
+import scala.math._
+import scala.reflect.ClassTag
+import scalaz.Scalaz._
+import scalaz._
+import scalaz.concurrent.Task
+
+/**
+  * Angular size of an image. Used to store the real size of images instead to the requested size
+  */
+case class AngularSize(ra: Angle, dec: Angle) {
+  def this(size: Angle) = this(size, size)
+
+  def halfDec = (dec / 2).getOrElse(Angle.zero)
+  def halfRa = (ra / 2).getOrElse(Angle.zero)
+}
+
+case object AngularSize {
+  val zero = AngularSize(Angle.zero, Angle.zero)
+
+  /** @group Typeclass Instances */
+  implicit val equal = Equal.equalA[AngularSize]
+
+  /** @group Typeclass Instances */
+  implicit val order: Order[AngularSize] =
+    Order.orderBy(a => (a.ra.toRadians * a.dec.toRadians))
```

Comment:
Maybe this doesn't matter (?) but I expected to see (ra, dec) instead of (ra * dec).  With the multiplication version two instances that are not equals could compare `EQ`.